### PR TITLE
feat: cleanup and stability improvements

### PR DESCRIPTION
**Pull request by ffMathy**
- [x] I have added automatic tests where applicable
- [x] The PR title is suitable as a release note
- [x] The PR contains a description of what has been changed
- [ ] The description contains manual test instructions

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "noEmit": true,
     "resolveJsonModule": true,
     "skipLibCheck": true,
-    "strict": false,
+    "strict": true,
```
Comment (gismya) [inline tsconfig.json:14]: If it still passes the tests it's a great change.

File: source/index.ts [hunk 2]
```diff
@@ -30,7 +30,10 @@ async function generate(
   const errors: unknown[] = [];
   let interfaces = "";
   for (const schema of schemas[0]) {
-    if (legacySchemas.includes(schema.id)) return;
+    if (legacySchemas.includes(schema.id)) {
+      return { errors, schemas };
```
[the remark this answers is not in the document]
Comment (ffMathy) [reply source/index.ts:34]: The problem is that if the return type of the whole function can be `{ errors, schemas } | undefined`, it becomes harder to handle the results.

Instead I'd rather let `errors` be empty, and `schemas` be empty. So we don't have to do null checks everywhere.

File: .yarnrc.yml [hunk 3]
```diff
@@ -5,3 +5,4 @@ plugins:
     spec: "@yarnpkg/plugin-interactive-tools"
 
 yarnPath: .yarn/releases/yarn-3.5.0.cjs
+nodeLinker: node-modules
```
[the remark this answers is not in the document]
Comment (ffMathy) [reply .yarnrc.yml:8]: Yeah, the `node_modules` folder didn't show up for me at all. Adding this fixed it when I ran `yarn install`.

File: tsconfig.json (hunk 1 above)
Comment (ffMathy) [reply tsconfig.json:14]: It sure does!

File: source/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (gismya) [reply source/index.ts:34]: I mean I don't think we should return at all inside the for-loop. Instead we should just get the rest of the for look inside 
`if (!legacySchemas.includes(schema.id))`. I think, I'm grabbing how everything is put together from memory.

[the remark this answers is not in the document]
Comment (ffMathy) [reply source/index.ts:34]: Ah so you mean the existing like that was already there was also buggy? Just need to make sure I understand.

[the remark this answers is not in the document]
Comment (gismya) [reply source/index.ts:34]: That's my interpretation of the code, yes.

File: .yarnrc.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (gismya) [reply .yarnrc.yml:8]: No, `pnp` (The default behaviour in recent yarn), doesn't use a `node_modules` folder. In theory it should be a lot faster with very few downsides, but unfortunately there are a lot of the rest of the node ecosystem that doesn't quite work well with `pnp`, so I almost always end up disabling it and using `node-modules`.

[the remark this answers is not in the document]
Comment (ffMathy) [reply .yarnrc.yml:8]: Ah, didn't know about pnp.

People are switching to bun.sh now either way. It's even faster than pnpm too.

File: source/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ffMathy) [reply source/index.ts]: I have now resolved it! 😄

**Review by gismya — APPROVED** [review]

**Comment by ffMathy** [comment]
What is missing for this to be merged? I think it could be a good idea to avoid conflicts.

**Comment by gismya** [comment]
I had simply missed pressing the button, sorry.

**Pull request merged**